arch/xtensa/src/esp32/esp32_rt_timer.c

```diff
@@ -216,24 +219,22 @@ static void stop_rt_timer(FAR struct rt_timer_s *timer)
 
       if (ishead)
         {
-          if (!list_is_empty(&s_runlist))
+          if (!list_is_empty(&priv->runlist))
             {
               /* Set the value from the next timer as the new hardware timer
                * alarm value
                */
 
-              next_timer = container_of(s_runlist.next,
+              next_timer = container_of(priv->runlist.next,
                                         struct rt_timer_s,
                                         list);
               alarm = next_timer->alarm;
 
-              ESP32_TIM_SETALRVL(tim, alarm);
-              ESP32_TIM_SETALRM(tim, true);
+              ESP32_TIM_SETALRVL(priv->timer, alarm);
+              ESP32_TIM_SETALRM(priv->timer, true);
             }
         }
     }
-
-  leave_critical_section(flags);
```

Comment:
Why did you remove the lock from the `stop_rt_timer` and `start_rt_timer`?
Can't we have an error in case one rt timer client call one of these 2 functions at the same time that another rt timer client does?

Reply:
Ok, I just saw in the end of the file you're externally protecting it.